packages/eslint-plugin/src/rules/lines-around-comment.ts

```diff
@@ -0,0 +1,528 @@
+            : parent;
+
+        return (
+          token.loc.start.line - parentStartNodeOrToken.loc.start.line === 1
+        );
+      }
+
+      return false;
+    }
+
+    /**
+     * Returns whether or not comments are at the parent end or not.
+     */
+    function isCommentAtParentEnd(
+      token: TSESTree.Token,
+      nodeType: TSESTree.AST_NODE_TYPES,
+    ): boolean {
+      const parent = getParentNodeOfToken(token);
+
+      return (
+        !!parent &&
+        isParentNodeType(parent, nodeType) &&
+        parent.loc.end.line - token.loc.end.line === 1
+      );
+    }
+
+    /**
+     * Returns whether or not comments are at the block start or not.
+     */
+    function isCommentAtBlockStart(token: TSESTree.Token): boolean {
+      return (
+        isCommentAtParentStart(token, AST_NODE_TYPES.ClassBody) ||
+        isCommentAtParentStart(token, AST_NODE_TYPES.BlockStatement) ||
+        isCommentAtParentStart(token, AST_NODE_TYPES.StaticBlock) ||
+        isCommentAtParentStart(token, AST_NODE_TYPES.SwitchCase)
+      );
+    }
+
+    /**
+     * Returns whether or not comments are at the block end or not.
+     */
+    function isCommentAtBlockEnd(token: TSESTree.Token): boolean {
+      return (
+        isCommentAtParentEnd(token, AST_NODE_TYPES.ClassBody) ||
+        isCommentAtParentEnd(token, AST_NODE_TYPES.BlockStatement) ||
+        isCommentAtParentEnd(token, AST_NODE_TYPES.StaticBlock) ||
+        isCommentAtParentEnd(token, AST_NODE_TYPES.SwitchCase) ||
+        isCommentAtParentEnd(token, AST_NODE_TYPES.SwitchStatement)
+      );
+    }
+
+    /**
+     * Returns whether or not comments are at the class start or not.
+     */
+    function isCommentAtClassStart(token: TSESTree.Token): boolean {
+      return isCommentAtParentStart(token, AST_NODE_TYPES.ClassBody);
+    }
+
+    /**
+     * Returns whether or not comments are at the class end or not.
+     */
+    function isCommentAtClassEnd(token: TSESTree.Token): boolean {
+      return isCommentAtParentEnd(token, AST_NODE_TYPES.ClassBody);
+    }
+
+    /**
+     * Returns whether or not comments are at the object start or not.
+     */
+    function isCommentAtObjectStart(token: TSESTree.Token): boolean {
+      return (
+        isCommentAtParentStart(token, AST_NODE_TYPES.ObjectExpression) ||
+        isCommentAtParentStart(token, AST_NODE_TYPES.ObjectPattern)
+      );
+    }
+
+    /**
+     * Returns whether or not comments are at the object end or not.
+     */
+    function isCommentAtObjectEnd(token: TSESTree.Token): boolean {
+      return (
+        isCommentAtParentEnd(token, AST_NODE_TYPES.ObjectExpression) ||
+        isCommentAtParentEnd(token, AST_NODE_TYPES.ObjectPattern)
+      );
+    }
+
+    /**
+     * Returns whether or not comments are at the array start or not.
+     */
+    function isCommentAtArrayStart(token: TSESTree.Token): boolean {
```

Comment:
The base rule already covers arrays, classes, and basically everything else that's not an interface or type. So you should be able to do what other rule extensions do and call the base rule to reuse its logic:

```ts
    const rules = baseRule.create(context, [options]);

    return {
      Program(): void {
        rules.Program();

        comments.forEach(...);
```

I prototyped this locally and it looks to get pretty much all the existing test cases. Is there any reason why this strategy wouldn't work?

It'd be a real shame to copy & paste so much core ESLint rule logic. We'd then have to maintain it all over time.

Reply:
Sorry for the late response! I finally have time to take a look at this.

Since the base rule will always report false positive errors on TS codes (which is what we're fixing here) and to reuse the `Program()` from the base rule, I create a custom `context.report` function that silences errors caused by TS-codes and 
 I feed it to the base rule.

Then I can also skip checking non-TS codes since they are already handled by the base rule.

Changes in [30825c6](https://github.com/typescript-eslint/typescript-eslint/pull/5327/commits/30825c6f8072ea09a2a63737ffdd78e844401be6)